From 3f1802aa5e9de4facac5a531d1a99b6a42d2936c Mon Sep 17 00:00:00 2001
From: Paul Cacheux <paul.cacheux@datadoghq.com>
Date: Tue, 10 May 2022 11:01:37 +0200
Subject: [PATCH] Export `LoadKernelSpecWithoutCache`

---
 btf/btf.go | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/btf/btf.go b/btf/btf.go
index 93b1c45..fa64aa1 100644
--- a/btf/btf.go
+++ b/btf/btf.go
@@ -13,7 +13,7 @@ import (
 	"reflect"
 	"sync"
 
-	"github.com/DataDog/btf-internals"
+	internal "github.com/DataDog/btf-internals"
 	"github.com/DataDog/btf-internals/sys"
 	"github.com/DataDog/btf-internals/unix"
 )
@@ -262,14 +262,14 @@ func LoadKernelSpec() (*Spec, error) {
 	}
 
 	var err error
-	kernelBTF.Spec, err = loadKernelSpec()
+	kernelBTF.Spec, err = LoadKernelSpecWithoutCache()
 	return kernelBTF.Spec, err
 }
 
-// loadKernelSpec attempts to load the raw vmlinux BTF blob at
+// LoadKernelSpecWithoutCache attempts to load the raw vmlinux BTF blob at
 // /sys/kernel/btf/vmlinux and falls back to scanning the file system
 // for vmlinux ELFs.
-func loadKernelSpec() (*Spec, error) {
+func LoadKernelSpecWithoutCache() (*Spec, error) {
 	fh, err := os.Open("/sys/kernel/btf/vmlinux")
 	if err == nil {
 		defer fh.Close()
-- 
2.36.1

